src/python/pants/util/meta.py

```diff
@@ -57,6 +57,33 @@ def __get__(self, obj: T, objtype: Optional[Type[T]] = None) -> Any:
         return callable_field()
 
 
+def runtime_subscriptable(cls: C) -> C:
```

Comment:
Maybe rename to include the word "ignore"? I continue to feel weird about this, but making it clear what will happen from the name would help.